packages/api/src/car.js

```diff
@@ -127,20 +134,50 @@ export async function handleCarUpload (request, env, ctx, car, uploadType = 'Car
   }
 
   const contentCid = normalizeCid(sourceCid)
+
+  // provide a pin object for elastic-ipfs.
+  const elasticPin = (structure) => ({
+    status: structure === 'Complete' ? 'Pinned' : 'Pinning',
+    contentCid,
+    location: {
+      peerName: 'elastic-ipfs',
+      // there isn't a seperate cluster peer for e-ipfs, so we reuse the ipfs peer.
+      peerId: env.ELASTIC_IPFS_PEER_ID,
+      ipfsPeerId: env.ELASTIC_IPFS_PEER_ID
+    }
+  })
+
   await env.db.createUpload({
     user: user._id,
     authKey: authToken?._id,
     contentCid,
     sourceCid,
     name,
     type: uploadType,
-    backupUrls: [`https://${env.s3BucketName}.s3.${env.s3BucketRegion}.amazonaws.com/${bucketKey}`],
-    pins: [],
+    backupUrls: [
+      `https://${env.s3BucketName}.s3.${env.s3BucketRegion}.amazonaws.com/${s3Key}` //,
+      // `https://${env.r2BucketName}.${env.accountId}.r2.cloudflarestorage.com/${r2bucketKey}`
```

Comment:
I'd like to stash a url for the r2 backup of the CAR here, but I'm not sure what URL to use. I'm guessing here based on some things I've seen in the r2 docs, but it doesn't look like the public url structure for r2 buckets is documented yet... also this would require us to duplicate our bucket config as env vars so we could get the "bucket  name" which is hidden from the worker normally as it just gets bound to an env var, and we'd also need an env var for the account id. Not a problem, but it makes our wrangler.toml a little gross, as the same values would need to be in two places.

Reply:
R2 public buckets did not land yet... So, there is really no way of getting it for now.

Reply:
You could always have a single `env.r2BucketUrl`.

I vote we put _something_ here - we should keep track of who uploaded what.

You could just make something up - all we'd need to do is create a worker to provide access:

```
prod-2.bucket.web3.storage
```

Reply:
I think we want to be able to map between CARs in s3 and CARs in r2 _somehow_. Perhaps we could use fictional url for our r2 bucket that we could turn into a real worker that serves things from r2 if we need to. We can't use the s3 url directly at the moment either, as we don't have public access enabled

```js
`http://carpark.web3.storage/${bucketKey}`
```

Reply:
DONE! I have made CARPARK_URL an env var.